dnf/base.py

```diff
@@ -1064,6 +1066,15 @@ def download_packages(self, pkglist, progress=None, callback_total=None):
             percent = 100 - real / full * 100
             logger.info(msg, full / 1024 ** 2, real / 1024 ** 2, percent)
 
+        if self.conf.destdir:
+            dnf.util.ensure_dir(self.conf.destdir)
+            for pload in payloads:
+                shutil.copy(pload.pkg.repo.pkgdir + "/" + os.path.basename(pload.pkg.location), self.conf.destdir)
+                if self.conf.destdir.endswith('/'):
```

Comment:
Permissions will be set to S_IRWXO is 777, so read and write for everyone, as you suggested. 
After shutil.copy, the permissions are read for everyone, write only for owner/group.
The main argument for 777 is that users will, most of the time, do something with the downloaded files - probably copy or move them. This action will most probably be done under an unprivileged user, so most users would end up switching to a privileged user to delete the downloaded files.
This poses no security risk, as all packages are checked for checksum and signature before installing.

Path joining fixed.